### ISSUE TITLE
Merge `twisted/internet/_endpointspy3.py` into `twisted/internet/endpoints.py`

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6236]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#6236|
|Type|task|
|Created|2012-12-30 19:22:10Z|
|Branch|https://github.com/twisted/twisted/tree/endpoints-merge-6236|

Also merge the associated test module. See [#6183](https://github.com/twisted/twisted/issues/6183).

<details><summary>Searchable metadata</summary>

```
trac-id__6236 6236
type__task task
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_endpoints_merge_6236 branches/endpoints-merge-6236
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1356895330000000 1356895330000000
changetime__1357008321000000 1357008321000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [36683]) Branching to 'endpoints-merge-6236'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/endpoints-merge-6236)

Once this lands, reviewing [#4696](https://github.com/twisted/twisted/issues/4696) will hopefully be easier.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_3"></a>|@tomprince set owner to @exarkun|
|-|-|

This seems to lack ```__future__``` imports. (The test module lacked it to begin with, but the python3 guide seems to suggest it be there anyway).

There are also a couple of new blank line errors reported by twistedchecker in ```test_endpoint```. All of the reported errors might as well be cleaned up now.

All of the changed code looks correct, and I did a quick scan of the copied code looking at shape, and things seem to be in order.

Please commit after fixing.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [36703]) Merge endpoints-merge-6236

Author: exarkun
Reviewer: tom.prince
Fixes: [#6236](https://github.com/twisted/twisted/issues/6236)

Re-integrate the Python 3 ported endpoints implementation into
`twisted/internet/endpoints.py`
